### PR TITLE
Fix _score access in double scripted terms agg

### PR DESCRIPTION
**Pull request by nik9000**
When we're aggregation on a script that contains `score` with the value
type hint `double` we were always using a score of `0.0`. This passes
the score into the script properly.

This also updates `terms` agg tests in accordance with #82273. I found
this bug while updating tests for #82273 but wanted to split out the bug
fix. But I wanted to include all of the updates to these tests for
issue #82273 because they add extra type checks for similar issues.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/fielddata/SortableLongBitsSortedNumericDocValues.java [hunk 1]
```diff
@@ -45,4 +47,11 @@ public int docValueCount() {
     public SortedNumericDoubleValues getDoubleValues() {
         return values;
     }
+
+    @Override
+    public void setScorer(Scorable scorer) {
+        if (values instanceof ScorerAware aware) {
+            aware.setScorer(scorer);
+        }
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/index/fielddata/SortableLongBitsSortedNumericDocValues.java:55]: This is the actual bug fix.

**Review by nik9000 — COMMENTED** [review]
I needed the dopamine fix from removing code. But I found this bug while I was doing it!

File: server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/bucket/DoubleTermsIT.java [hunk 2]
```diff
@@ -127,7 +127,7 @@ public void setupSuiteScopeCluster() throws Exception {
                     .setSource(
                         jsonBuilder().startObject()
                             .field(SINGLE_VALUED_FIELD_NAME, (double) i)
-                            .field("num_tag", i < NUM_DOCS / 2 + 1 ? 1 : 0) // used to test order by single-bucket sub agg
+                            .field("num_tag", i < NUM_DOCS / 2 + 1 ? 1.0 : 0.0) // used to test order by single-bucket sub agg
```
Comment (not-napoleon) [inline server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/bucket/DoubleTermsIT.java:130]:
```suggestion
                            .field("num_tag", i < (NUM_DOCS / 2) + 1 ? 1.0 : 0.0) // used to test order by single-bucket sub agg
```

File: server/src/main/java/org/elasticsearch/index/fielddata/SortableLongBitsSortedNumericDocValues.java (hunk 1 above)
Comment (not-napoleon) [reply server/src/main/java/org/elasticsearch/index/fielddata/SortableLongBitsSortedNumericDocValues.java:55]: ooo, the new instance of syntax.  Very nice.

**Review by not-napoleon — APPROVED** [review]
LGTM, pending tests passing.

**Comment by nik9000** [comment]
@elasticmachine, update branch

**Comment by nik9000** [comment]
run  elasticsearch-ci/part-1

**Comment by nik9000** [comment]
Thanks for the review @not-napoleon !

**Pull request merged**